### PR TITLE
some minor tests added

### PR DESCRIPTION
**Pull request by mac2000**
Hello, while trying to figure out how everything works did tried to implement few tests based on coverage results

### CODE REVIEW

**Comment by SimonCropp** [comment]
@sungam3r LGTM

File: src/GraphQL.Tests/NameFieldResolverTests.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@ public class NameFieldResolverTests
         [InlineData("naMe", "Anyone")]
         [InlineData("FullInfo", "Anyone 20")]
         [InlineData("fullInfo", "Anyone 20")]
+        [InlineData(null, null)]
```
Comment (joemcbride) [inline src/GraphQL.Tests/NameFieldResolverTests.cs]: This looks like a case that should never happen.  You should never actually have a FieldAst with a null name.

Comment (mac2000) [reply src/GraphQL.Tests/NameFieldResolverTests.cs]: Understood, my intent was only to cover edge case, and to prove that it wont blowup with null ref exception, if you wish we can remove this

Comment (sungam3r) [reply src/GraphQL.Tests/NameFieldResolverTests.cs]: This test is OK.

File: src/GraphQL.Tests/TypeExtensionsTests.cs [hunk 2]
```diff
@@ -32,6 +33,12 @@ public void is_nullable_using_nullable_types(Type type)
             TypeExtensions.IsNullable(type).ShouldBeTrue();
         }
 
+        [Fact]
+        public void will_throw_on_unknown_list()
```
Comment (sungam3r) [inline src/GraphQL.Tests/TypeExtensionsTests.cs]:
```suggestion
        public void will_throw_on_open_generic()
```

File: src/GraphQL.Tests/TypeExtensionsTests.cs [hunk 3]
```diff
@@ -32,6 +33,12 @@ public void is_nullable_using_nullable_types(Type type)
             TypeExtensions.IsNullable(type).ShouldBeTrue();
         }
 
+        [Fact]
+        public void will_throw_on_open_generic()
```
Comment (sungam3r) [inline src/GraphQL.Tests/TypeExtensionsTests.cs]: And one more - please move this test to src\GraphQL.Tests\Utilities\TypeExtensionsTests.cs

Comment (sungam3r) [reply src/GraphQL.Tests/TypeExtensionsTests.cs]: You also mind to rename it to GetGraphTypeFromTypeTests.cs to avoid ambiguity.

File: src/GraphQL.Tests/Utilities/FederatedSchemaBuilderTests.cs [hunk 4]
```diff
@@ -120,5 +121,34 @@ public void entity_query()
                 _.ExpectedResult = expected;
             });
         }
+
+        [Fact]
+        public void input_types_and_types_without_key_directive_are_not_added_to_entities_union()
```
Comment (sungam3r) [inline src/GraphQL.Tests/Utilities/FederatedSchemaBuilderTests.cs]: @joemcbride I haven’t had time yet to sort out federated schemas. Does this test seem right to you?

File: src/GraphQL.Tests/TypeExtensionsTests.cs (hunk 3 above)
Comment (mac2000) [reply src/GraphQL.Tests/TypeExtensionsTests.cs]: I did move test and rename file, the only thing I did not touch all other tests, this file has half tests named in snake case and half in something between pascal and snake case

Comment (sungam3r) [reply src/GraphQL.Tests/TypeExtensionsTests.cs]: OK

File: src/GraphQL.Tests/TypeExtensionsTests.cs [hunk 5]
```diff
@@ -1,4 +1,5 @@
 using System;
+using System.Collections.Generic;
```
Comment (sungam3r) [inline src/GraphQL.Tests/TypeExtensionsTests.cs]:
```suggestion
```

Comment (mac2000) [reply src/GraphQL.Tests/TypeExtensionsTests.cs]: Oops, sorry for that, my bad
